docs/fsharp/style-guide/formatting.md

```diff
@@ -116,657 +62,512 @@ This is sometimes called “vanity alignment” or “vanity indentation”. The
+x ||| y // Bitwise or, also for working with “flags” enumeration
+x &&& y // Bitwise and, also for working with “flags” enumeration
+x ^^^ y // Bitwise xor, also for working with “flags” enumeration
 ```
 
-## Formatting records
+## Formatting if expressions
 
-Short records can be written in one line:
+Indentation of conditionals depends on the size and complexity of the expressions that make them up.
+Write them on one line when:
+
+- `cond`, `e1`, and `e2` are short
+- `e1` and `e2` are not `if/then/else` expressions themselves.
 
 ```fsharp
-let point = { X = 1.0; Y = 0.0 }
+if cond then e1 else e2
 ```
 
-Records that are longer should use new lines for labels:
+If any of the expressions are multi-line or `if/then/else` expressions.
 
 ```fsharp
-let rainbow =
-    { Boss = "Jeffrey"
-      Lackeys = ["Zippy"; "George"; "Bungle"] }
+if cond then
+    e1
+else
+    e2
 ```
 
-Placing the opening token on a new line, the contents tabbed over one scope, and the closing token on a new line is preferable if you are:
-
-* Moving records around in code with different indentation scopes
-* Piping them into a function
+Multiple conditionals with `elif` and `else` are indented at the same scope as the `if` when they follow the rules of the one line `if/then/else` expressions.
 
 ```fsharp
-let rainbow =
-    {
-        Boss1 = "Jeffrey"
-        Boss2 = "Jeffrey"
-        Boss3 = "Jeffrey"
-        Boss4 = "Jeffrey"
-        Boss5 = "Jeffrey"
-        Boss6 = "Jeffrey"
-        Boss7 = "Jeffrey"
-        Boss8 = "Jeffrey"
-        Lackeys = ["Zippy"; "George"; "Bungle"]
-    }
+if cond1 then e1
+elif cond2 then e2
+elif cond3 then e3
+else e4
+```
 
-type MyRecord =
-    {
-        SomeField: int
-    }
-    interface IMyInterface
+If any of the conditions or expressions is multi-line, the entire `if/then/else` expression is multi-line:
 
-let foo a =
-    a
-    |> Option.map (fun x ->
-        {
-            MyField = x
-        })
+```fsharp
+if cond1 then
+    e1
+elif cond2 then
+    e2
+elif cond3 then
+    e3
+else
+    e4
 ```
 
-The same rules apply for list and array elements.
-
-## Formatting copy-and-update record expressions
+If a condition is long, the `then` keyword is still placed at the end of the expression.
 
-A copy-and-update record expression is still a record, so similar guidelines apply.
+```fsharp
+if
+    complexExpression a b && env.IsDevelopment()
+    || secondLongerExpression
+        aVeryLongparameterNameOne
+        aVeryLongparameterNameTwo
+        aVeryLongparameterNameThree then
+        e1
+    else
+        e2
+```
 
-Short expressions can fit on one line:
+It is, however, better style to refactor long conditions to a let binding or separate function:
 
 ```fsharp
-let point2 = { point with X = 1; Y = 2 }
+let performAction =
+    complexExpression a b && env.IsDevelopment()
+    || secondLongerExpression
+        aVeryLongparameterNameOne
+        aVeryLongparameterNameTwo
+        aVeryLongparameterNameThree
```

Comment:
Is there an extra space in these params?

Reply:
No, I don't think so - they are parameters to a function